tianshou/utils/net/common.py

```diff
@@ -138,14 +150,26 @@ def __init__(
 
     @no_type_check
     def forward(self, obs: Union[np.ndarray, torch.Tensor]) -> torch.Tensor:
-        if self.device is not None:
-            obs = torch.as_tensor(obs, device=self.device, dtype=torch.float32)
+        obs = torch.as_tensor(obs, device=self.device, dtype=torch.float32)
```

Comment:
why change this?

Reply:
A little less complexity and more consistency. I think `self.device` should take precedence over any devices that may have been declared for the tensors earlier, do you agree? Otherwise here you could get observations on the GPU and the model on cpu (with device None)